src/commands.ts

```diff
@@ -640,6 +644,22 @@ export const insertFootnote = (
     ).pop()
 
     if (footnoteElement) {
+      if (isDeleted(footnoteElement.node)) {
+        const footnoteElementPos =
+          footnotesSection.pos + findFootnotePos(footnotesSection.node)
```

Comment:
No need for this `findFootnotePos`. You already have the `footnotesSection.pos` & the `footnoteElement.pos`, please use these values to find the `footnoteElementPos`.

Reply:
I've already tried this, but it returned an incorrect position

Reply:
Weird! The `findFootnotePos` function refers to the paragraph inside a footnote position. How does it work to refer to the footnote element position? 🤔

Reply:
This may be working because we only have one footnote inside the footnotes element, but as far as I can see, it will fail if we have more than one. I suggest changing it to: `const footnoteElementPos = footnotesSection.pos + footnoteElement.pos + 1`. It seems to be off by one for some reason, but I'm not sure why.

Reply:
It did. The "findFootnotePos" function first locates the footnote element by identifying the footnote node and capturing its position within the "footnoteSection". It then traverses the footnote node to find a paragraph inside (as the paragraph is the core content we're concerned with) and adds its relative position to the footnote element's position. If no paragraph is found, it defaults to the start of the footnote element, ensuring it correctly refers to the element's position.

Reply:
@IslamJMomani 

> This may be working because we only have one footnote inside the footnotes element, but as far as I can see, it will fail if we have more than one. I suggest changing it to: `const footnoteElementPos = footnotesSection.pos + footnoteElement.pos + 1`. It seems to be off by one for some reason, but I'm not sure why.

What do you mean by "it will fail if we have more than one"? Are you referring to having more than one footnote before deleting the footnote element, or are you talking about adding multiple footnotes after undoing the deletion of a footnote element? Please check the attached record; it works fine when inserting multiple footnotes after undoing the deletion of the footnote element: 
[Screencast from 09-24-2024 12:35:17 PM.webm](https://github.com/user-attachments/assets/1385f8db-62b4-4f4c-819e-05b5c5207dda)

Reply:
I mean, when we fix the bug in LEAN-4013, we will have a case of multiple deleted footnotes inside a deleted footnotes element. So when we try to insert a footnote, the `findFootnotePos` function will incorrectly mark the last footnote position as the footnotes element position, which is wrong.

Reply:
believe addressing LEAN-4013 may lead to several changes, and this could be one of them. so maybe many things will be changed as well